src/components/PopupWithForm.js

```diff
@@ -6,8 +6,28 @@ export default class PopupWithForm extends Popup {
     this.submit = submitFunction;
     this.form = this.modal.querySelector(".form");
     this.inputs = this.form.querySelectorAll(".form__input");
+    this.button = this.form.querySelector(".form__submit-button");
+    this.buttonText = this.button.textContent;
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      if (this.form.id === "form-delete-card") {
+        this.button.textContent = "Deleting...";
+      } else {
+        this.button.textContent = "Saving...";
+      }
+    } else {
+      this.button.textContent = this.buttonText;
+    }
+  }
+
+  setUpDeleteModal = (cardObj) => {
+    this.currentId = cardObj.id;
+    this.removeHandler = cardObj._removeCard;
+    super.openModal();
```

Comment:
Needs correction: setting up the delete handler and opening the modal are different actions

Reply:
Still needs corrections